### DISCUSSION TITLE
How to run the project?

### DISCUSSION THREAD

**Discussion by CocaColf** [laike9m/Cyberbrain#115] [Q&A] [answered]
It's seems a fool question but I had tried to find solution.

I'm a font-end engineer and no experience about Python, but I am interested in this project(I knew it a long time ago, from GitHub and Podcast "PythonHunter").

I cloned the project and use `poetry` and `npm` to install dependences.

And then I start to find how to run it.

Firstly, I spent little time to read `cyberbrain-vsc ` code, and I guess maybe there is an local server, and draw graph by  data from the server.

Secondly, The usage of Cyberbrain is: `we can decorate a function with `@trace`,then run the program`. So I think maybe I should install the local `cyberbrain` code as an Python package? Just like [npm link](https://docs.npmjs.com/cli/v7/commands/npm-link).
But I don't find `poetry` has the same feature.

Thirdly, I run  `python cyberbrain/__init__.py`, but get an error:

![image](https://user-images.githubusercontent.com/25732253/116531465-c59ef800-a911-11eb-95bb-7ce1395c79c2.png)

I don’t know how to do it now, so I come to seek your help. I hope you can understand my poor English :joy:

**Comment by laike9m** [comment, marked as answer]
Hi @CocaColf, thank you for your interest in the project. FYI, we have a [Chinese discussion](https://discord.gg/fKR7P2sf) channel on Discord if you'd like to join.

To answer your question. It depends on whether you want to **use Cyberbrain** or **develop it**.

- If you want to use Cyberbrain, just follow the guidance [here](https://github.com/laike9m/Cyberbrain#install) to install the Python package and VS Code extension.

- If you want to develop it, we have a [development guide](https://github.com/laike9m/Cyberbrain/blob/master/docs/Development.md). It does not cover how Cyberbrain works though, for which I'll write another doc.

Hope that helps.

**Reply by CocaColf** [reply]
> Hi @CocaColf, thank you for your interest in the project. FYI, we have a [Chinese discussion](https://discord.gg/fKR7P2sf) channel on Discord if you'd like to join.
> 
> To answer your question. It depends on whether you want to **use Cyberbrain** or **develop it**.
> 
> * If you want to use Cyberbrain, just follow the guidance [here](https://github.com/laike9m/Cyberbrain#install) to install the Python package and VS Code extension.
> * If you want to develop it, we have a [development guide](https://github.com/laike9m/Cyberbrain/blob/master/docs/Development.md). It does not cover how Cyberbrain works though, for which I'll write another doc.
> 
> Hope that helps.

Thanks for your reply! I originally feel is's a nice project and wanted to contribute to it. So I belong to those who want to develop it.

I'm looking forward to the doc coming!

**Reply by laike9m** [reply]
OK. If you want to develop it, the development guide should get you going. If you're able to run all the tests following the instructions, then it shows everything is working.

I just added a [section](https://github.com/laike9m/Cyberbrain/blob/master/docs/Development.md#run-the-extension-under-development) to show how to run it. Let me know if you have any questions.
